File: Python/Maximum Subarray/MaxSubarray.py
class Solution:
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        maxnum = nums[0]
        cursor = nums[0]
        for num in nums[1:]:
            cursor = num if cursor < 0 else cursor + num
            if (cursor > maxnum):
                maxnum = cursor
        return maxnum

File: C#/Maximum Subarray/MaxSubarray.cs
public class Solution {
    public int MaxSubArray (int[] nums) {
        int[] sub = new int[nums.Length];
        int max = nums[0];
        sub[0] = max;
        for (int i = 1; i < nums.Length; i++) {
            sub[i] = nums[i] + sub[i - 1] > nums[i] ? nums[i] + sub[i - 1] : nums[i];
            if (max < sub[i])
                max = sub[i];
        }
        return max;
    }
}